lua/neogit/popups/ignore/actions.lua

```diff
@@ -0,0 +1,58 @@
+local M = {}
+
+---@param path string
+---@param rules string[]
+local function add_rules(path, rules)
+  local async = require("plenary.async")
+  local notification = require("neogit.lib.notification")
+
+  local err, fd = async.uv.fs_open(path, "a+", 438)
+  if err then
+    return notification.error(string.format("Failed to read ignore file %q\n\n%s", path, err))
+  end
+
+  async.uv.fs_write(fd, table.concat(rules, "\n") .. "\n")
+
+  local err = async.uv.fs_close(fd)
+  if err then
+    return notification.error(string.format("Failed to close file %q", path))
+  end
```

Comment:
Is there an advantage to using this low-level API versus the plenary Path abstraction? I had a lot of success in the past replacing these type of calls with Path.

Reply:
No, the only one I can think of is error handling, rather than the asserts inside `Path:write`.

How do you go about the docs for plenary.path? I usually read the source file, but it is easy to miss things that way.